wire-library/wire-runtime-swift/src/test/swift/StringEncodedTests.swift

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright 2020 Square Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import Foundation
+import XCTest
+@testable import Wire
+
+final class StringEncodedTests: XCTestCase {
+    struct SupportedTypes : Codable, Equatable {
+        @StringEncoded
+        var a: Int64
+        @StringEncoded
+        var b: UInt64
+        @StringEncodedValues
+        var c: [Int64]
+        @StringEncodedValues
+        var d: [UInt64]
+        @StringEncoded
+        var e: Int64?
+        @StringEncoded
+        var f: Int64?
+        @StringEncoded
+        var g: UInt64?
+        @StringEncoded
+        var h: UInt64?
+        @DefaultEmpty
+        @StringEncodedValues
+        var i: [Int64]
+        @DefaultEmpty
+        @StringEncodedValues
+        var j: [UInt64]
+    }
+
+    func testSupportedTypes() throws {
```

Comment:
This was a copy/pasted and modified version of JSONStringTests.
The big difference is starting with `inputJson` and ensuring that the expectedStruct can be created with a minimal payload.

Reply:
- [ ] add throw assertion tests